aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,459 @@
+            error -> fail());
+
+        verify(mobileClient, times(1))
+                .confirmSignUp(any(String.class), any(String.class), any(Callback.class));
+    }
+
+    /**
+     * Tests that the resendSignUpCode method of the Auth wrapper of AWSMobileClient (AMC) calls AMC.resendSignUp with
+     * the username it received .
+     * Also ensures that in the onResult case, the success callback receives a valid AuthSignUpResult.
+     */
+    @Test
+    public void resendSignUpCode() {
+        SignUpResult amcResult = new SignUpResult(
+                false,
+                new UserCodeDeliveryDetails(
+                        destination,
+                        deliveryMedium,
+                        attributeName
+                ),
+                null
+        );
+
+        doAnswer(invocation -> {
+            assertEquals(username, invocation.getArgument(0));
+
+            Callback<SignUpResult> callback = invocation.getArgument(1);
+            callback.onResult(amcResult);
+            return null;
+        }).when(mobileClient)
+            .resendSignUp(any(String.class), any(Callback.class));
+
+        authPlugin.resendSignUpCode(
+            username,
+            result -> validateSignUpResult(result, AuthSignUpStep.CONFIRM_SIGN_UP_STEP),
+            error -> fail());
+
+        verify(mobileClient, times(1))
+                .resendSignUp(any(String.class), any(Callback.class));
+    }
+
+    /**
+     * Tests that the signIn method of the Auth wrapper of AWSMobileClient (AMC) calls AMC.signIn with
+     * the username, password it received, and, if included, the metadata sent in the options object.
+     * Also ensures that in the onResult case, the success callback receives a valid AuthSignInResult.
+     */
+    @Test
+    public void signIn() {
+        SignInResult amcResult = new SignInResult(
+            SignInState.SMS_MFA,
+            new UserCodeDeliveryDetails(
+                    destination,
+                    deliveryMedium,
+                    attributeName
+            )
+        );
+
+        doAnswer(invocation -> {
+            assertEquals(username, invocation.getArgument(0));
+            assertEquals(password, invocation.getArgument(1));
+            assertEquals(metadata, invocation.getArgument(2));
+
+            Callback<SignInResult> callback = invocation.getArgument(3);
+            callback.onResult(amcResult);
+            return null;
+        }).when(mobileClient)
+            .signIn(any(String.class), any(String.class), any(Map.class), any(Callback.class));
+
+        authPlugin.signIn(
+            username,
+            password,
+            AWSCognitoAuthSignInOptions.builder().metadata(metadata).build(),
+            result -> validateSignInResult(
+                result,
+                false,
+                AuthSignInStep.CONFIRM_SIGN_IN_WITH_SMS_MFA_CODE
+            ),
+            error -> fail());
```

Comment:
To my earlier point about needing to make these calls synchronous -- this is why our other components all have a corresponding test utility called like `SynchronousWhatever`. It allows us to write easy jUnit tests that perform actions in a serial way. I would suggest you make like a `SynchronousAuth` utility, which wraps the `AuthCategoryBehavior`. Then, you could either do `SynchronousAuth.delegatingTo(authCategory)`, or `SynchronousAuth.delegatingTo(authPlugin)`, since both are examples of the `AuthCategoryBehavior`.

Reply:
Will sync up with you